lib/png/png.go

```diff
@@ -0,0 +1,119 @@
+package png
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	_ "embed"
+
+	"github.com/playwright-community/playwright-go"
+	"oss.terrastruct.com/d2/lib/xmain"
+)
+
+type Playwright struct {
+	PW      *playwright.Playwright
+	Browser playwright.Browser
+	Page    playwright.Page
+}
+
+func (pw *Playwright) RestartBrowser() (newPW Playwright, err error) {
+	if err = pw.Browser.Close(); err != nil {
+		return Playwright{}, err
+	}
+	return startPlaywright(pw.PW)
+}
+
+func (pw *Playwright) Cleanup() error {
+	if err := pw.Browser.Close(); err != nil {
+		return err
+	}
+	if err := pw.PW.Stop(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func startPlaywright(pw *playwright.Playwright) (Playwright, error) {
+	browser, err := pw.Chromium.Launch()
+	if err != nil {
+		return Playwright{}, err
+	}
+	context, err := browser.NewContext()
```

Comment:
Why do you need the context? Why not `browser.NewPage`?

Reply:
In the Playwright docs it says
```
**This is a convenience API that should only be used for the single-page scenarios and short snippets. Production code and testing frameworks should explicitly create [browser.newContext([options])](https://playwright.dev/docs/api/class-browser#browser-new-context) followed by the [browserContext.newPage()](https://playwright.dev/docs/api/class-browsercontext#browser-context-new-page) to control their exact life times.**
```

https://playwright.dev/docs/api/class-browser#browser-new-page